eslint.config.mjs

```diff
@@ -0,0 +1,183 @@
+import { includeIgnoreFile } from '@eslint/compat';
+import {
+  baseSpecRules,
+  createBaseConfig,
+  createDocsConfig,
+  createTestConfig,
+  EXTENSION_TEST_FILE,
+  EXTENSION_TS,
+} from '@mui/internal-code-infra/eslint';
+import { defineConfig } from 'eslint/config';
+import * as path from 'node:path';
+import { fileURLToPath } from 'url';
+
+const filename = fileURLToPath(import.meta.url);
+const dirname = path.dirname(filename);
+
+const OneLevelImportMessage = [
+  'Prefer one level nested imports to avoid bundling everything in dev mode or breaking CJS/ESM split.',
+  'See https://github.com/mui/material-ui/pull/24147 for the kind of win it can unlock.',
+].join('\n');
+
+const NO_RESTRICTED_IMPORTS_PATTERNS_DEEPLY_NESTED = [
+  {
+    group: ['@base-ui-components/react/*/*'],
+    message: OneLevelImportMessage,
+  },
+];
+
+// Add relevant packages to the list below.
+const NO_RESTRICTED_IMPORTS_PATHS_TOP_LEVEL_PACKAGES = [
+  // { name: string, message: string }
+];
+
+export default defineConfig(
+  includeIgnoreFile(path.join(dirname, '.gitignore')),
+  includeIgnoreFile(path.join(dirname, '.eslintignore')),
+  {
+    name: 'Base Config',
+    extends: createBaseConfig({
+      baseDirectory: dirname,
+    }),
+    settings: {
+      'import/resolver': {
+        typescript: {
+          project: ['tsconfig.json'],
+        },
+      },
+    },
+    /**
+     * Sorted alphanumerically within each group. built-in and each plugin form
+     * their own groups.
+     */
+    rules: {
+      'import/export': 'off', // Mostly handled by Typescript itself. ESLint produces false positives with declaration merging.
+      'no-restricted-imports': [
+        'error',
+        {
+          patterns: NO_RESTRICTED_IMPORTS_PATTERNS_DEEPLY_NESTED,
+        },
+      ],
+      '@typescript-eslint/no-redeclare': 'off',
+      // We LOVE non-breaking spaces, and both straight and curly quotes here
```

Comment:
Why not relax this on the company level?

Reply:
Noted for next PR in `mui-public`.